### DISCUSSION TITLE
Kubernetes cannot scrape the operator pod (tries port 80, but the operator uses port 8080)?

### DISCUSSION THREAD

**Discussion by kastl-ars** [microsoft/retina#734] [Q&A] [answered]
Hi all,

I just installed Retina and kube-prometheus-stack according to the documentation.

But one of the retina-pods targets is unhealthy according to Prometheus. It tries to get the metrics for the operator pod on port 80:

```
Get "http://10.42.0.16:80/metrics": dial tcp 10.42.0.16:80: connect: connection refused
```

However, the operator pods seems to listen on port 8080?

```
[...]
ts=2024-09-12T11:23:41.024Z level=info caller=legacy/deployment.go:251 msg="Starting manager"
2024-09-12T11:23:41.024Z        info    controller-runtime.metrics      Starting metrics server
2024-09-12T11:23:41.024Z        info    controller-runtime.metrics      Serving metrics server  {"bindAddress": ":8080", "secure": false}
2024-09-12T11:23:41.024Z        info    starting server {"name": "health probe", "addr": "[::]:8081"}
[...]
```

This is the values.yaml I uses for the installation:

```
image:
  tag: 'v0.0.16' 
operator:
  enabled: true
  tag: 'v0.0.16' 
  enableRetinaEndpoint: true
loglevel: 'info'
enabledPlugin_linux: "[dropreason,packetforward,linuxutil,dns,packetparser]"
enablePodLevel: true
remoteContext: true
```

Any ideas?

Kind Regards,
Johannes

**Comment by SRodi** [comment]
@mereta

**Comment by mereta** [comment, marked as answer]
Hey @kastl-ars!

Thanks for opening this discussion! 
We've created an issue and recently pushed a fix.

The operator didn't have the correct configuration set up for Prometheus to scrape metrics.
We have added a ServiceMonitor called retina-operator-servicemonitor, which when enabled will scrape metrics from retina-operator.

![image](https://github.com/user-attachments/assets/7e83dd49-8f8c-459c-8fcf-c6086b885d5e)

To enable the monitor you should append a clause in your values.yaml file.
```
metrics:
   serviceMonitor:
      enabled: true
```

You can find more details in this PR #848 

Thanks :)

**Reply by kastl-ars** [reply]
Thanks a lot for the fix!
